### PR TITLE
Add null-aware access to semantics instance

### PR DESCRIPTION
**Pull request by htoor3**
We are making calls on a `SemanticsTextEditingStrategy.instance` that might be null in certain cases inside testing environments.  We need to add conditional property access to these instances.

Fixes https://github.com/flutter/flutter/issues/122131

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by htoor3** [comment]
@yjbanov @mdebbar should we merge this for now to unblock `samehere` as we think of a better way to ensure these instances are initialized in testing environments?

File: lib/web_ui/lib/src/engine/semantics/text_field.dart [hunk 1]
```diff
@@ -33,14 +33,14 @@ class SemanticsTextEditingStrategy extends DefaultTextEditingStrategy {
...
-      return instance;
+    if (_instance != null && instance?.owner == owner) {
+      return instance!;
     }
     return _instance = SemanticsTextEditingStrategy(owner);
   }
 
   /// The [SemanticsTextEditingStrategy] singleton.
-  static SemanticsTextEditingStrategy get instance => _instance!;
+  static SemanticsTextEditingStrategy? get instance => _instance!;
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/semantics/text_field.dart:43]: I would suggest keeping this non-nullable (i.e. remove the `?`).

And callsites should be updated to use the private nullable `_instance`:
```dart
SemanticsTextEditingStrategy._instance?.activate(this);
SemanticsTextEditingStrategy._instance?.deactivate(this);
```

**Review by mdebbar — APPROVED** [review]

File: lib/web_ui/test/engine/semantics/text_field_test.dart [hunk 2]
```diff
@@ -43,7 +43,7 @@ void testMain() {
     setUp(() {
       testTextEditing = HybridTextEditing();
       SemanticsTextEditingStrategy.ensureInitialized(testTextEditing);
-      strategy = SemanticsTextEditingStrategy.instance;
+      strategy = SemanticsTextEditingStrategy.instance!;
```
Comment (mdebbar) [inline lib/web_ui/test/engine/semantics/text_field_test.dart:46]: Not needed anymore.

File: lib/web_ui/test/engine/semantics/text_field_test.dart [hunk 3]
```diff
@@ -446,7 +446,7 @@ void testMain() {
...
```
Comment (mdebbar) [inline lib/web_ui/test/engine/semantics/text_field_test.dart:449]: same here

**Review by mdebbar — APPROVED** [review]
Ship it!

**Comment by htoor3** [comment]
@mdebbar thoughts on whether I should add to this PR the change we discussed a few days ago on this line https://github.com/flutter/engine/blob/89627fda305517674ec68edc3501a8707eaf2d4f/lib/web_ui/lib/src/engine/semantics/semantics.dart#L1789 - where we move that conditional out? 

Are there are situations in which semantics gets enabled before we hit `updateSemantics` and then it goes through that code path when it shouldn't be?

File: lib/web_ui/lib/src/engine/semantics/text_field.dart [hunk 4]
```diff
@@ -460,6 +460,6 @@ class TextField extends RoleManager {
     if (!isIosSafari) {
       editableElement?.remove();
     }
-    SemanticsTextEditingStrategy.instance.deactivate(this);
+    SemanticsTextEditingStrategy._instance?.deactivate(this);
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/semantics/text_field.dart:463]: Is it possible to add an engine test that simulates the situation where dispose is called before the strategy is initialized?

**Comment by mdebbar** [comment]
> @mdebbar thoughts on whether I should add to this PR the change we discussed a few days ago

It's up to you, but I would put it in a separate PR so it can be reverted independently in case it breaks something.

File: lib/web_ui/test/engine/semantics/text_field_test.dart [hunk 5]
```diff
@@ -36,6 +36,33 @@ void testMain() {
...
+      semantics()
+        ..debugOverrideTimestampFunction(() => _testTime)
+        ..semanticsEnabled = true;
+    });
+
+    tearDown(() {
+      semantics().semanticsEnabled = false;
+    });
+
+    test('Calling dispose() pre-initialization will not throw an error', () {
```
Comment (htoor3) [inline lib/web_ui/test/engine/semantics/text_field_test.dart:50]: @yjbanov would something like this work?

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**